docs/development.rst

```diff
@@ -268,6 +268,120 @@ active, as long as the Python `tox` package is installed in it:
     $ tox -e py27                      # Run tests on Python 2.7
 
 
+.. _`Testing installed versions of pywbem`:
+
+Testing installed versions of pywbem
+------------------------------------
+
+By default, the tests use the pywbem and pywbem_mock modules from the
+respective directories in the Git repo work directory.
+
+Pywbem 0.14.5 introduced a way to test installed versions of the pywbem
+package. This is useful for example for testing a version of pywbem that has
+been packaged as an OS-level package. Typically, such a version would be
+installed into the system Python.
+
+Some words of caution:
+
+* Testing an installed version of pywbem with test cases from a pywbem repo
+  of a different version can result in failing test cases for several reasons:
+
+  - If a new version of pywbem has added functionality, its test cases are also
+    extended accordingly. Running such newer test cases against an older
+    installed version of pywbem may fail simply because the installed version
+    does not yet have the added functionality.
+
+  - Fixes in pywbem or in the test cases may change behavior in a subtle way
+    that causes test cases to fail.
+
+  - Unit test cases are particularly vulnerable to version mismatches because
+    they test at the module level, including module interfaces that are
+    internal to pywbem and thus can legally change incompatibly between
+    versions.
+
+* If the version of the installed pywbem is before 0.14.5, some test cases
+  that compile MOF will be skipped to avoid permission denied errors when
+  ply attempts to re-generate its parsing table files in the pywbem
+  installation directory in case of ply version mismatches. Starting with
+  pywbem 0.14.5, it has tolerance against ply version mismatches.
+
+In order to not clutter up the system Python with Python packages needed for
+running the pywbem tests, the following steps use a virtual Python environment
+that uses the packages of the system Python. That way, the installed version of
+pywbem becomes available to the virtual Python environment from the system
+Python, while any additional packages that are needed but not yet available
+that way, will be installed into the virtual Python environment.
+
+Follow these steps to run pywbem tests against a version of pywbem that is
+installed into the system Python:
+
+1. Verify that the following commands are available when the system Python
+   is active:
+
+   ::
+
+       $ virtualenv --version   # Python virtualenv package
+       $ pip --version
+
+2. Create and activate a virtual Python environment of the intended Python
+   version, that is based on the system Python:
+
+   ::
+
+       $ virtualenv --system-site-packages --no-setuptools --no-pip --no-wheel .virtualenv/test
+       $ source .virtualenv/test/bin/activate
+
+   The pywbem project is set up so that Git ignores the ``.virtualenv``
```

Comment:
Question: The use of the .virtualenv as a dir name. Is that not user/env specific.  Thus, when I setup virtualenv and virtualenvwrapper it sets .virtualenvs as the the name and virtualenvwrapper uses that

WORKON_HOME="/home/kschopmeyer/virtualenvs"

Reply:
NOISSUE - WORKON_HOME is used only by virtualenvwrapper which is on top of virtualenv. virtualenvwrapper is not really useable in scripts, so scripts need to use virtualenv. virtualenv gets the directory specified directly, which is the case in the example code shown.